kernel/timeout.c

```diff
@@ -73,7 +73,17 @@ static s32_t next_timeout(void)
 
 #ifdef CONFIG_TIMESLICING
 	if (_current_cpu->slice_ticks && _current_cpu->slice_ticks < ret) {
+#ifndef CONFIG_SMP
```

Comment:
Why we cannot just set slice_ticks = 0 for idle thread?

Reply:
> Why we cannot just set slice_ticks = 0 for idle thread?

@pizi-nordic thanks for the review, yes, changing time slicing related variables is more reasonable. But my trigger point is in timeout.c, I don't want to modify that variables outside sched.c. Maybe we should add more interfaces in sched.c to enable/disable time slicing for kernel side without changing slice_time. Thanks again.

Reply:
My fault: I mistaken _current_cpu with _current :).
What about changing condition to:
```
if ((_current_cpu->slice_ticks && _current_cpu->slice_ticks < ret && sliceable(_current))
```
?

Reply:
@pizi-nordic thanks. I think sliceable(_idle_thread) equals to my first condition (_current != _idle_thread), but we have to consider another case, _current == _idle_thread && _kernel.ready_q.cache != _idle_thread, this is because when _current is _idle_thread and a external interrupt or a timer interrupt wakeup a real thread, we have to resume the time slicing function timely. And I think my changes "(_current != _idle_thread || _kernel.ready_q.cache != _idle_thread) " can cover above concerns. Thanks

Reply:
Also, most likely, this change just add this "_current != _idle_thread" overhead at run time.

Reply:
I do not like the way time slicing is implemented as we have to workaround bugs created by poor design.

The is reset_time_slice() called when ready cache is updated and this sets the timeout resuming slicing operation. But then timer interrupt might fire and overwrite this information. I am not sure if this is the scenario you have in mind, but I see why you have to check ready queue.

Reply:
> The is reset_time_slice() called when ready cache is updated and this sets the timeout resuming slicing operation. But then timer interrupt might fire and overwrite this information. I am not sure if this is the scenario you have in mind, but I see why you have to check ready queue.

yes, that's my consideration. Thanks.